google/cloud/storage/oauth2/google_credentials.h

```diff
@@ -0,0 +1,78 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_OAUTH2_GOOGLE_CREDENTIALS_H_
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_OAUTH2_GOOGLE_CREDENTIALS_H_
+
+#include "google/cloud/internal/throw_delegate.h"
+#include "google/cloud/storage/oauth2/anonymous_credentials.h"
+#include "google/cloud/storage/oauth2/authorized_user_credentials.h"
+#include "google/cloud/storage/oauth2/service_account_credentials.h"
+#include "google/cloud/storage/version.h"
+#include <memory>
+#include <utility>
+
+namespace google {
+namespace cloud {
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+namespace oauth2 {
+
+/**
+ * Produces a credential type based on a the runtime environment. If the
```

Comment:
s/a the/the/

Keep the first sentence into its own paragraph so Doxygen will pick it up as the brief description:

```C++
/**
 * Produces a credential type based on the runtime environment.
 *
 * If the GOOGLE_APPLICATION_CREDENTIALS...
```